demo/components/checkbox-demo.tsx

```diff
@@ -3,50 +3,156 @@ import {observable} from 'mobx'
 
 const style = require('./checkbox-demo.st.css').default;
 
 @observer
-export class CheckBoxDemo extends React.Component<{},{value: boolean}> {
+export class CheckBoxDemo extends React.Component<{},{}> {
+    render() {
+        return (
+            <div>
+                <div>
+                    <h3>Basic CheckBox</h3>
+                    <BasicDemo/>
+                </div>
+
+                <div>
+                    <h3>Disabled</h3>
+                    <DisabledDemo/>
+                </div>
+
+                <div>
+                    <h3>Indeterminate</h3>
+                    <IndeterminateDemo/>
+                </div>
+
+                <div>
+                    <h3>Custom Icons</h3>
+                    <CustomIconsDemo/>
+                </div>
+            </div>
+        )
+    }
+}
 
-    @observable value: boolean = demoCheckBoxInitialValue;
+@observer
+class BasicDemo extends React.Component<{}, {}> {
+    @observable value: boolean = false;
 
-    handleChange = (val: boolean) => {this.value = val};
+    handleChange = (val: boolean) => { this.value = val };
 
     render() {
         return (
-            <div>
+            <div data-automation-id="BASIC_DEMO">
                 <CheckBox value={this.value}
-                          text={demoCheckBoxText}
-                          boxIcon={CheckBoxSVG}
-                          tickIcon={TickMarkSVG}
-                          onChange={this.handleChange}/> <br/>
+                          onChange={this.handleChange}>
+                    <span data-automation-id="BASIC_LABEL" className={style.label}>{demoCheckBoxText}</span>
+                </CheckBox> <br/>
                 <button disabled={!this.value} data-automation-id="BUTTON_SUBMIT">
-                    Submit
+                    Proceed
                 </button>
             </div>
         )
     }
 }
 
-const CheckBoxSVG: React.SFC<CheckBoxIconProps> = (props) => {
+@observer
+class DisabledDemo extends React.Component<{}, {}> {
+    @observable value: boolean = false;
+
+    handleChange = (val: boolean) => { this.value = val };
+
+    render() {
+        return (
+            <div>
+                <span data-automation-id="DISABLED_DEMO">
+                    <CheckBox value={this.value}
+                              onChange={this.handleChange}
+                              disabled>
+                        <span data-automation-id="DISABLED_LABEL" className={style.label}>Unchecked</span>
+                    </CheckBox>
+                </span>
+                <span>
+                    <CheckBox value={true} disabled>
+                        <span className={style.label}>Checked</span>
+                    </CheckBox>
+                </span>
+            </div>
+        )
+    }
+}
+
+@observer
+class IndeterminateDemo extends React.Component<{}, {}> {
+    @observable values: { option1: boolean, option2: boolean } = {option1: true, option2: false};
```

Comment:
I think this would be easier to read if we had two separate properties for value1 and value2, with inline event handlers for both: `onChange={newVal => this.value1 = newVal}`